modules/gdscript/gdscript.cpp

```diff
@@ -2639,12 +2639,34 @@ Ref<GDScript> GDScriptLanguage::get_script_by_fully_qualified_name(const String
 
 /*************** RESOURCE ***************/
 
+Ref<Resource> ResourceFormatLoaderGDScript::load_cyclic(const String &p_path, Error *r_error) {
+	Error err = OK;
+	Ref<Resource> res = GDScriptCache::get_shallow_script(p_path, err);
+	if (err != OK) {
+		*r_error = err;
+	}
+	return res;
+}
+
 Ref<Resource> ResourceFormatLoaderGDScript::load(const String &p_path, const String &p_original_path, Error *r_error, bool p_use_sub_threads, float *r_progress, CacheMode p_cache_mode) {
 	if (r_error) {
 		*r_error = ERR_FILE_CANT_OPEN;
 	}
 
-	Error err;
+	Error err = OK;
+	FileAccess::open(p_path, FileAccess::READ, &err);
+	if (err != OK) {
+		if (r_error) {
+			*r_error = err;
+		}
+		return Ref<GDScript>();
+	}
+	if (r_error) {
+		// As we could open the file, we can assume that any error opening the file is linked to a cyclic reference
+		*r_error = ERR_CYCLIC_LINK;
+	}
```

Comment:
This seems wrong, it will be overridden below by
```
	if (r_error) {
		*r_error = OK;
	}
```

Reply:
Semantically, it's also weird to set an error in an external variable *before* actually attempting to load the script which may not at all be an error. It's less descriptive for the reader, and could lead to bugs if there are race conditions?

Reply:
I see pre-existing code does the same though with
```
	if (r_error) {
		*r_error = ERR_FILE_CANT_OPEN;
	}
```

It's a bit weird.
Normally we'd use a local `Error err` to keep track of the error, and only set `r_error` before returning (on failure, or at the end of the function on success).

Reply:
All loaders use a pointer for errors instead of editing the reference itself. It's because it's editing the value of the `ResourceLoader::ThreadLoadTask` struct `error` value.

See this example:
https://github.com/godotengine/godot/blob/629796c333bcc46f2aeb4399c1a5786d6b013289/core/io/image_loader.cpp#L150-L157